### PR TITLE
Use VFS as service account issuer if configured

### PR DESCRIPTION
**Pull request by olemarkus**
Think this is the closest path to both a test and getting rid of the PublicJWKS feature flag when using VFS.

@johngmyers don't remember if we concluded we wanted a flag to explicitly state if the user wants VFS or API server for OIDC, or if we just want to infer that like we do right now.

### CODE REVIEW

**Comment by johngmyers** [comment]
My recollection was to infer.

Since we learned that newer k8s can have multiple service account issuers, we might just want to add them all. That requires figuring out a backwards-compatible API change, though–perhaps a second `serviceAccountIssuers` field which takes a slice?

File: cmd/kops/integration_test.go [hunk 1]
```diff
@@ -291,21 +291,30 @@ func TestPrivateDns2(t *testing.T) {
...
 	// We have to use a fixed CA because the fingerprint is inserted into the AWS WebIdentity configuration.
-	newIntegrationTest("minimal.example.com", "public-jwks").
+	newIntegrationTest("minimal.example.com", "public-jwks-apiserver").
 		withCAKey().
 		withServiceAccountRole("dns-controller.kube-system", true).
 		runTestTerraformAWS(t)
 
 }
 
+func TestPublicJWKSVFS(t *testing.T) {
```
Comment (johngmyers) [inline cmd/kops/integration_test.go:309]: TestVFSServiceAccountIssuerDiscovery?

**Comment by olemarkus** [comment]
> My recollection was to infer.
> 
> Since we learned that newer k8s can have multiple service account issuers, we might just want to add them all. That requires figuring out a backwards-compatible API change, though–perhaps a second `serviceAccountIssuers` field which takes a slice?

We still should only configure one AWS OIDC Provider. But we may say we only support using the VFS for that. Same with IRSA for our own addons.

**Comment by johngmyers** [comment]
Ah, I was talking about what goes into componentconfig. But this takes more research. Later PR; infer VFS if configured for now.

**Review by johngmyers — APPROVED** [review]

**Pull request merged**